sympy/control/core.py

```diff
@@ -0,0 +1,269 @@
+                    "are currently supported')
+        if T is None:
+            T = basis[0]
+        else:
+            T = T.row_join(basis[0])
+    T.simplify()
+    return similarity_transformation(ss, T), T
+
+
+def tf2ss(G):
+    """
+    Convert a transfer function or transfer function matrix to 
+    a state-space represenation using a Gilbert minimal realization.
+    """
+    if not hasattr(G, 'shape'):
+        G = sympy.Matrix([G])
+    # find unique roots
+    roots = []
+    for i in range(G.shape[0]):
+        for j in range(G.shape[1]):
+            for t in G[i, j].apart(s).as_ordered_terms():
+                for root in sympy.roots(sympy.denom(t), s):
+                    if root not in roots:
+                        roots += [root]
+
+    n = len(roots)
+    m = G.shape[0]
+    p = G.shape[1]
+    A = sympy.zeros(n)
+    C = None
+    B = None
+    D = G.applyfunc(lambda x: x.apart(s).limit(s, sympy.oo))
+
+    # for each root
+    for i_root, root in enumerate(roots):
+
+        # create a from root
+        A[i_root, i_root] = root
+
+        # find residue
+        G_r = G.applyfunc(
+            lambda x:
+                sympy.residue(x, s, root).simplify())
+        u_syms = sympy.symbols('u0:{:d}'.format(m))
+        v_syms = sympy.symbols('v0:{:d}'.format(p))
+        u = sympy.Matrix(u_syms)
+        v = sympy.Matrix(list(v_syms))
+        vals = {key: 1 for key in list(u) + list(v)}
+        sol = sympy.solve(
+            list(u*v.T - G_r),
+            u_syms + v_syms, dict=True)[0]
+        vals.update(sol) 
+        vals = {
+            key: sympy.sympify(vals[key]).subs(vals)
+            for key in vals.keys()}
+        if C is None:
+            C = u.subs(vals)
+        else:
+            C = C.row_join(u.subs(vals))
+
+        if B is None:
+            B = v.subs(vals).T
+        else:
+            B = B.col_join(v.subs(vals).T)
+    return StateSpace(A, B, C, D, 0)
+
+
+def ctrb(ss):
+    """
+    Compute the controllability matrix
+
+    :param ss: The state space
+    :return : O (controllability matrix)
+    """
+    n = ss.A.shape[0]
+    O_ctrb = ss.B
+    for i in range(n-1):
+        O_ctrb = O_ctrb.row_join(ss.A**n*ss.B)
+    return O_ctrb
+
+
+def obsvb(ss):
+    """
+    Compute the observability matrix
+
+    :param ss: The state space
+    :return : O (observability matrix)
+    """
+
+    if n is None:
+        n = ss.A.shape[0]
+    O_ctrb = ss.C
+    for i in range(n-1):
+        O_ctrb = O_ctrb.row_join(ss.C*ss.A**n)
+    return O_ctrb
+
+def pade(f, n):
+    """
+    Pade approximation
+
+    :param f: A function, cannot contain symbols.
+    :n : The order of the numerator and
+        denominator polynomial for the pade
+        approximation.
+    """
+    a = mpmath.taylor(f, 0, 2*n + 1)
+    num, den = mpmath.pade(a, n, n)
+    num.reverse()
+    den.reverse()
+    return sympy.nsimplify(
+        tf(num, den)).simplify()
```

Comment:
Do not use `simplify()` in library code. Functions should return unsimplified forms and users decide if they want to make use of `simplify()`.